joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForExpressionCreator.scala

```diff
@@ -75,11 +78,43 @@ trait AstForExpressionCreator(implicit withSchemaValidation: ValidationMode) { t
     Seq(callAst(cNode, operand))
   }
 
+  private def astForIndexExpression(indexNode: ParserNodeInfo): Seq[Ast] = {
+    val indexAst                                = astForNode(indexNode.json(ParserKeys.Index))
+    val (indexIdentifier, callNodeTypeFullName) = processIndexIdentifier(indexNode.json(ParserKeys.X))
+    val callNode =
+      createCallNodeForOperator(indexNode, Operators.indexAccess, typeFullName = Some(callNodeTypeFullName))
+    Seq(callAst(callNode, indexIdentifier ++ indexAst))
+  }
+
+  private def processIndexIdentifier(identNode: Value): (Seq[Ast], String) = {
+    val identifierAst = astForNode(identNode)
+    val identifierTypeFullName =
+      identifierAst.head.root.get.properties
+        .get(PropertyNames.TYPE_FULL_NAME)
+        .getOrElse(Defines.anyTypeName)
+        .toString
+        .stripPrefix("*")
+        .stripPrefix("[]")
+    (identifierAst, identifierTypeFullName)
+  }
+//
+//  private def extractBaseType(input: String): String = {
```

Comment:
Remember to remove commented out code or attach a note if you still want to keep it